src/pages/index.js

```diff
@@ -65,55 +45,94 @@ function handleEditProfileButtonClick() {
   formValidators["profile-form"].disableSubmitButton();
 }
 
-profilePopup.setEventListeners();
+function loadApiCards() {//no longer used save as a part of the "emergency option" so holding on to just in case
+  api.getCards().then(data => {
+    data.forEach((datum) => (renderCard(datum)));
+  });
+}
 
-editProfileButton.addEventListener("click", handleEditProfileButtonClick);
+function restoreOriginalCards() {//in case I ever go nuclear on the cards again by accident... call on standby at the bottom
+  initialCards.forEach((card) => {
+    api.postCard(card);    
+  });
+  loadApiCards();
+}
+
+function handleDelButtonClick(card) {
+  deleteDialogue.open();
+  deleteIdentification = card;
+}
 
-function renderCard(card) {//to address the recursive code I had been writing in the section and addphoto initializations
-  const newCard = new Card(card, "#card", handlePhotoClick);
+function renderCard(card) {
+  const newCard = new Card(card, "#card", handlePhotoClick, handleDelButtonClick, handleLikeClick);//ammended the delete and like handler so it could tell the other popup to open when the Card's del button was pressed and utilize the api.  
   const newCardElement = newCard.generateCardElement();
   cardSection.addItem(newCardElement);
 }
 
-const addPopup = new PopupWithForm(addPhotoPopup, {
-  handleFormSubmit: ({ title, image_url: imageUrl }) => {
-    const newObj = {
-      name: title,
-      link: imageUrl
-    };
-    renderCard(newObj);    
-    formValidators["add-photo-form"].resetFormValidation();
-    addPopup.close();
-  },
-});
-
-
-addPopup.setEventListeners();
-
-const cardSection = new Section(
-{
-  items: initialCards,
-  renderer: renderCard,
-},
-".cards"
-);
-
-cardSection.renderItems();
+function handleDeleteSubmit() {//did this as a pass-to for the delete popup just because I was having such a difficult time workshopping how exactly to delete, needed to segregate it. Came up with this.
+  api.deleteCard(deleteIdentification.id)
+  .then(() => {
+    deleteDialogue.renderLoading(true, "Deleting...");    
+    deleteIdentification.deleteElement();//I guess I was storing the to-be-deleted card all along and just needed to access it. WOW.
+    formValidators["delete-photo-form"].disableSubmitButton();
+  })
+  .then(() => {
+    setTimeout(() => {
```

Comment:
Hey there
Part of why I was still using the timeouts was, in part, because my internet is decent so all the requests were pretty quick- to even see if the renderLoading function was working or not I needed to stretch it out artificially so I could at least see the changes happening in the submit button. I guess I don't need it anymore on those grounds now that I know it's working; I also know if some hypothetical user has a less optimal/slower internet connection that issue sort of solves itself, and in fact might be exacerbated by a timeout on a successful api call- do you think it would it be a user experience problem if the timer were still left in for the .then blocks? (As in, "oh I hit the submit and it's thinking... man this is taking a long time.") Tell me what you think about it. I still think it would be valid to have them in .catch blocks so the user would have some notification that something had gone awry and could see it but I will defer to your judgement.

Reply:
> I still think it would be valid to have them in .catch blocks

You don't need `timeout` because the main reason of showing the `Saving...` text is to let the user know that the request is being processed otherwise the user may think that it didn't work and will try to click the button again and again